nodes/core/parsers/70-CSV.js

```diff
@@ -199,17 +194,25 @@ module.exports = function(RED) {
                             o[node.template[j]] = k[j];
                         }
                         if (JSON.stringify(o) !== "{}") { // don't send empty objects
-                            if (node.multi === "one") {
-                                var newMessage = RED.util.cloneMessage(msg);
-                                newMessage.payload = o;
-                                node.send(newMessage); // either send
-                            }
-                            else { a.push(o); } // or add to the aray
+                            a.push(o); // add to the aray
                         }
                         if (node.multi !== "one") {
                             msg.payload = a;
                             node.send(msg); // finally send the array
                         }
+			else {
+			    var len = a.length;
+			    for(var i = 0; i < len; i++) {
+                                var newMessage = RED.util.cloneMessage(msg);
+                                newMessage.payload = a[i];
```

Comment:
re in case the existing message already has parts - we "push" them down onto a stack... - eg 

```
if (newMessage.hasOwnProperty("parts")) { newMessage.parts = { parts:newMessage.parts }; } // push existing parts to a stack                
else { newMessage.parts = {}; }
```

Reply:
Thank you for reviewing.
I changed CSV node to preserve incoming parts property.